src/libcore/iter/iterator.rs

```diff
@@ -98,6 +98,7 @@ fn _assert_is_object_safe(_: &dyn Iterator<Item=()>) {}
     message="`{Self}` is not an iterator"
 )]
 #[doc(spotlight)]
+#[must_use]
```

Comment:
This should be:
```suggestion
#[must_use = "iterator adaptors are lazy and do nothing unless consumed"]
```
(which can be fixed in a follow up PR)